src/main/java/jalf/relation/algebra/Union.java

```diff
@@ -67,4 +69,10 @@ protected RelationType typeCheck() {
         return visitor.visit(this);
     }
 
+
+    @Override
+    protected Keys lazyComputeKey() {
+        return left.getKeys().union(right.getKeys());
```

Comment:
No key is preserved: take the largest possible key (the one with all attributes)